### ISSUE TITLE
Activated round robin sharding algorithm but I still see sharding per k8s cluster

### ISSUE THREAD

**Issue by paolocarta** [argoproj/argo-cd#15466]
Checklist:

* [x ] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x ] I've included steps to reproduce the bug.
* [x ] I've pasted the output of `argocd version`.

**Describe the bug**

I use ArgoCD with sharding for the application controller. I have four k8s clusters and 4 app controller shards. After enabling the round robin algorithm I still see apps assigned by cluster after restarting the controllers. 

**To Reproduce**

Set the param controller.sharding.algorithm to round-robin. Restart the controllers and watch the debug level logs. You will see the usual legacy sharding algorithm, where apps are distributed by cluster.

**Expected behavior**

I would expect the apps assigned in a round robin way to the app controller shards.

**Version**

2.8.2

**Logs**

`{"level":"debug","msg":"ClustersList has 4 items","time":"2023-09-12T14:28:59Z"}
{"level":"debug","msg":"Adding cluster with id= and name=in-cluster to cluster's map","time":"2023-09-12T14:28:59Z"}
{"level":"debug","msg":"Adding cluster with id=6d88fc37-0749-46af-a796-57f2b085622c and name=bacei-prod to cluster's map","time":"2023-09-12T14:28:59Z"}
{"level":"debug","msg":"Adding cluster with id=aa4ed2a1-e8d3-40fc-a8bd-90a40aad3f12 and name=bacei-test to cluster's map","time":"2023-09-12T14:28:59Z"}
{"level":"debug","msg":"Adding cluster with id=cfecc607-8238-4e66-be75-7c66cc2feb50 and name=bacei-dev to cluster's map","time":"2023-09-12T14:28:59Z"}
{"level":"debug","msg":"Cluster with id=cfecc607-8238-4e66-be75-7c66cc2feb50 will be processed by shard 3","time":"2023-09-12T14:28:59Z"}
{"level":"debug","msg":"Cluster with id= will be processed by shard 0","time":"2023-09-12T14:28:59Z"}
{"level":"debug","msg":"Cluster with id=6d88fc37-0749-46af-a796-57f2b085622c will be processed by shard 1","time":"2023-09-12T14:28:59Z"}
{"level":"debug","msg":"Cluster with id=aa4ed2a1-e8d3-40fc-a8bd-90a40aad3f12 will be processed by shard 2","time":"2023-09-12T14:28:59Z"}`

**Comment by jannfis** [comment]
Hi @paolocarta , sharding always works on the cluster level, not on the application level.

Round-robin in this case means, that the clusters are distributed equally among the shards. Cluster 1 goes to first shard, cluster 2 to next shard, and so on.

**Comment by bygui86** [comment]
@jannfis thanks for the clarification. Is there any plan to support round robin sharding based on Applications instead of Clusters?

**Comment by jannfis** [comment]
@bygui86 Yes, sharding on application level is indeed something we want to evaluate going forward. Not sure if it would make sense in a round-robin fashion, or if some more intelligence would be required for that, though.

**Comment by bygui86** [comment]
@jannfis ok thanks for the clarification. Looking forward the sharding at application level :)

**Comment by rumstead** [comment]
Closing this as the question has been answered. Feel free to reopen but here is more information regarding application sharding on this issue https://github.com/argoproj/argo-cd/issues/22499
